lib/publiccloud/provider.pm

```diff
@@ -384,22 +385,26 @@ sub terraform_apply {
     record_info('TFM cmd', $cmd);
 
     script_retry($cmd, timeout => $terraform_timeout, delay => 3, retry => 6);
-    my $ret = script_run('terraform apply -no-color -input=false myplan', $terraform_timeout);
-    unless (defined $ret) {
-        if (is_serial_terminal()) {
-            type_string(qq(\c\\));    # Send QUIT signal
+    my $tries = 1;
+    $tries = 3 if (is_azure());    # Retry on azure platform
+    while (1) {
```

Comment:
```suggestion
    while ($ret) {
```

and obviously you need to define variable before in such a case . or can switch to `do {} while($ret)` so condition will be checked after first loop

Reply:
:+1: